src/presenter/film-presenter.js

```diff
@@ -0,0 +1,100 @@
+import {render, remove, replace} from '../framework/render.js';
+import {isEscPressed} from '../utils/common.js';
+import {pageBody} from '../const.js';
+import FilmCardView from '../view/film-card-view.js';
+import PopupView from '../view/popup-view.js';
+
+const Mode = {
+  DEFAULT: 'DEFAULT',
+  POPUP: 'POPUP',
+};
+
+export default class FilmCardPresenter {
+  #filmCardContainer = null;
+  #changeData = null;
+  #clearPopup = null;
+
+  #filmCardComponent = null;
+  #filmPopupComponent = null;
+
+  #filmCard = null;
+  #comments = null;
+  #mode = Mode.DEFAULT;
+
+  constructor(filmCardContainer, comments, changeData, clearPopup) {
+    this.#filmCardContainer = filmCardContainer;
+    this.#comments = comments;
+    this.#changeData = changeData;
+    this.#clearPopup = clearPopup;
+  }
+
+  init = (film) => {
+    this.#filmCard = film;
+
+    const prevFilmCardComponent = this.#filmCardComponent;
+
+    this.#filmCardComponent = new FilmCardView(film);
+
+    this.#filmCardComponent.setClickHandler(this.#renderPopup);
+    this.#filmCardComponent.setControlButtonClickHandler(this.#handleWatchlistClick, this.#handleWatchedClick, this.#handleFavoriteClick);
+
+    if (prevFilmCardComponent === null) {
+      render(this.#filmCardComponent, this.#filmCardContainer);
+      return;
+    }
+
+    if (this.#filmCardContainer.contains(prevFilmCardComponent.element)) {
+      replace(this.#filmCardComponent, prevFilmCardComponent);
+    }
+
+    remove(prevFilmCardComponent);
+  };
+
+  resetView = () => {
+    if (this.#mode !== Mode.DEFAULT) {
+      this.#removePopupFromPage();
+    }
+  };
+
+  #handleWatchlistClick = () => {
+    this.#changeData({...this.#filmCard, userDetails: { watchlist: !this.#filmCard.userDetails.watchlist }});
+  };
```

Comment:
Здесь ты сбрасываешь остальные чекбоксы. Нужна такая запись
```suggestion
  #handleWatchlistClick = () => {
    this.#changeData({...this.#filmCard, userDetails: { ...this.#filmCard.userDetails, watchlist: !this.#filmCard.userDetails.watchlist }});
  };
```

Ну и дальше так же